### DISCUSSION TITLE
Неправильно подключил

### DISCUSSION THREAD

**Discussion by NissSonn** [vvip-68/GyverPanelWiFi#263] [Задавайте вопросы] [answered]
Всем привет.
При тесте случайно (вот не знаю как так получилось) подключил матрицу, подав на DIN "ноль", с виду ничего страшного не произошло, а часть светодиодов загорелось красным.
Осознав ошибку переподключил всё правильно, но матрица признаков жизни не подаёт.
Собственно вопрос: Матрица, всё, мертва?

**Comment by vvip-68** [comment, marked as answer]
Вряд ли. Скорее всего либо где-то сделали ошибку при монтаже, либо выбрана не та версия ядра или плата при компиляции из за чего вывод идет не на тот пин, к которому реально подключена матрица.

**Comment by 247760** [comment]
Выживших легко проверить, залив такой простой скетч, им можно искать сгоревшие или не в ту сторону запаянные светодиоды, когда со старта всё не заработало с первого раза. В том или по за том году писал про это, прошу @vvip-68  расписать этот момент в мануале (перед этим проверив, а то мало ли) для тех у кого собранная лента не работает или горят кракозябры или горит пол ленты, чтобы исключить нерабочие светодиоды\ не в ту сторону запаянные и копать уже дальше прошивку. Может это уже и расписано, я не проверял, так как по памяти настраиваю.

**Бегущая радуга (microLED)**

> #define STRIP_PIN 2     // пин ленты
#define NUMLEDS 100     // кол-во всех светодиодов в испытуемой ленте
#include <microLED.h>   // подключаем библу
microLED < NUMLEDS, STRIP_PIN, -1, LED_WS2812, ORDER_GRB > strip;
void setup() {
  strip.setBrightness(60);  // яркость
}
void loop() {
  static byte counter = 0;
  for (int i = 0; i < NUMLEDS; i++) {
    strip.set(i, mWheel8(counter + i * 255 / NUMLEDS));   // counter смещает цвет
  }
  counter += 3;   // counter имеет тип byte и при достижении 255 сбросится в 0
  delay(30);      // 30 кадров в секунду
}

**Бегущая радуга (FastLED)**

> #define NUM_LEDS 144  // кол-во всех светодиодов в испытуемой ленте
#include "FastLED.h"
#define PIN 6
CRGB leds[NUM_LEDS];
byte counter;
void setup() {
  FastLED.addLeds<WS2811, PIN, GRB>(leds, NUM_LEDS).setCorrection( TypicalLEDStrip );
  FastLED.setBrightness(50);
  pinMode(13, OUTPUT);
}
void loop() {
  for (int i = 0; i < NUM_LEDS; i++ ) {         // от 0 до первой трети
    leds[i] = CHSV(counter + i * 2, 255, 255);  // HSV. Увеличивать HUE (цвет)
    // умножение i уменьшает шаг радуги
  }
  counter++;        // counter меняется от 0 до 255 (тип данных byte)
  FastLED.show();
  delay(5);         // скорость движения радуги
}

**Comment by NissSonn** [comment]
Спасибо.
Матрица выжила.
Видимо подвели меня стандартные контакты у самой матрицы, провод данных отвалился, а я мучился и понять не мог где косяк. А может, что-то ещё было не так.
Переделал и перепаял всё ещё раз и всё заработало.
